realm/src/main/java/io/realm/Realm.java

```diff
@@ -1477,6 +1487,7 @@ public static void migrateRealm(RealmConfiguration configuration, RealmMigration
             realm.beginTransaction();
             realm.setVersion(realmMigration.execute(realm, realm.getVersion()));
             realm.commitTransaction();
+            validatedRealmFiles.remove(configuration.getPath());
```

Comment:
`validatedRealmFiles.add()` is done within a write transaction. Should `remove()` also be done within a write transaction?